pkg/clusterversion/refer.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+//
+
+package clusterversion
+
+// Refer allows tying one or more objects (or even a code comment, etc)
```

Comment:
I wonder if we should rename this to `RequiredBy` or `UsedBy` something that makes it clearer that the thing is required for compatibility with the named version (and is able to be deleted when the named version is deleted)